content/blog/svglite-2-0-0/index.Rmd

```diff
@@ -0,0 +1,250 @@
+svglite is a graphic device that is capable of creating SVG files from R graphics.
+SVG is a vector graphic format which means that it encode the instructions for recreating a graphic in a scale-independent way.
+This is in contrast with raster graphics such as PNG (as can be produced with the graphic devices in [ragg](https://ragg.r-lib.org)) which encodes actual pixel values and will get pixelated as you zoom in.
+
+You can install the latest release of svglite from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("svglite")
+```
+
+Much time has passed since svglite had a major release and this blog post will go into detail with all the major changes.
+You can see a full list of changes in the [release notes](%7B%20github_release%20%7D).
+
+## Motivation for svglite
+
+As part of this release we have sharpened our personal motivation for creating and maintaining svglite in the face of the already existing `svg()` device provided by R.
+All of the changes that are part of this release somehow plays into these motivations and are thus grouped by it below.
+
+### Speed
+
+The defining feature at the inception of svglite was speed (as compared to `svg()`).
+At some point in the past this regressed quite considerably and this release not only brings it back to past glory, but improves further upon it.
+The combined effect of this is that you'll experience a 13x speed improvement (based on the speed test in the [readme](https://svglite.r-lib.org/index.html#speed)) using the new version over v1.2.3.
+svglite is now \~3x faster than using `svg()` based on the same benchmark:
+
+```{r, fig.asp=0.4, message = FALSE}
+library(tidyverse)
+library(svglite)
+
+x <- runif(1e3)
+y <- runif(1e3)
+tmp1 <- tempfile()
+tmp2 <- tempfile()
+
+svglite_test <- function() {
+  svglite(tmp1)
+  plot(x, y)
+  dev.off()
+}
+svg_test <- function() {
+  svg(tmp2, onefile = TRUE)
+  plot(x, y)
+  dev.off()
+}
+
+bench::mark("svglite()" = svglite_test(), 
+            "svg()" = svg_test(), 
+            min_iterations = 250) %>% 
+  plot(type = "ridge") + 
+  labs(x = NULL, y = NULL)
+```
+
+Speed is not the be-all-end-all of graphic devices but for a device based on a web-native file format it is quite important as it allows web servers based on e.g. shiny to stream dynamically created plots with low latency.
+
+The additional speed is mainly achieved by no longer ensuring that the file is valid svg *while* it is being rendered.
+If you need this you can get the old behavior (and speed) back by setting `always_valid = TRUE` when opening the device.
+You should however consider using `svgstring()` instead of `svglite()` if you need to continuously access the svg while it is being rendered.
+
+### File size
+
+For more or less the same reasons as given above for rendering speed, the final file size of the rendered SVG can be very important when streaming graphics from a web server.
+Smaller files result in lower latency.
```

Comment:
Similarly you can delete; I don't think you need to sell people on why smaller files are better.